src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs

```diff
@@ -1,345 +0,0 @@
-
-                var setupInfo = new AppDomainSetup();
-                setupInfo.ApplicationBase = Path.GetDirectoryName(Path.GetFullPath(source));
-
-                // In Dev10 by devenv uses its own app domain host which has default optimization to share everything.
-                // Set LoaderOptimization to MultiDomainHost which means:
-                //   Indicates that the application will probably host unique code in multiple domains, 
-                //   and the loader must share resources across application domains only for globally available (strong-named) 
-                //   assemblies that have been added to the global assembly cache. 
-                setupInfo.LoaderOptimization = LoaderOptimization.MultiDomainHost;
-
-                AppDomain ad = null;
-                try
-                {
-                    ad = AppDomain.CreateDomain("Dependency finder domain", null, setupInfo);
-
-                    var assemblyLoadWorker = typeof(AssemblyLoadWorker);
-                    AssemblyLoadWorker worker = null;
-                    if (assemblyLoadWorker.Assembly.GlobalAssemblyCache)
-                    {
-                        worker = (AssemblyLoadWorker)ad.CreateInstanceAndUnwrap(
-                            assemblyLoadWorker.Assembly.FullName,
-                            assemblyLoadWorker.FullName,
-                            false, BindingFlags.Default, null,
-                            null, null, null);
-                    }
-                    else
-                    {
-                        // This has to be LoadFrom, otherwise we will have to use AssemblyResolver to find self.
-                        worker = (AssemblyLoadWorker)ad.CreateInstanceFromAndUnwrap(
-                            assemblyLoadWorker.Assembly.Location,
-                            assemblyLoadWorker.FullName,
-                            false, BindingFlags.Default, null,
-                            null, null, null);
-                    }
-
-                    return worker.CheckAssemblyReference(source, referenceAssemblyName, referenceAssemblyPublicKeyToken);
-                }
-                finally
-                {
-                    if (ad != null)
-                    {
-                        AppDomain.Unload(ad);
-                    }
-                }
-            }
-            catch
-            {
-                return null; // Return null if something goes wrong.
-            }
-        }
-
-        /// <summary>
-        /// Finds platform and .Net framework version for a given test container.
-        /// If there is an error while infering this information, defaults (AnyCPU, None) are returned
-        /// for faulting container.
-        /// </summary>
-        /// <param name="testSource"></param>
-        /// <returns></returns>
-        public static KeyValuePair<Architecture, FrameworkVersion> GetFrameworkVersionAndArchitectureForSource(string testSource)
```

Comment:
I believe this was used by testwindow. Not sure it relevant anymore.

Reply:
why is testwindow depedant on objectmodel?

Reply:
[AssemblyMetadataProvider](https://github.com/Microsoft/vstest/blob/master/src/vstest.console/CommandLine/AssemblyMetadataProvider.cs) does the same work as this method?